### PR TITLE
Feature/vmo 6851/vmo 7102/fixes and corrections

### PR DESCRIPTION
**Pull request by safydy-r**
[Overview]
Validation on resources (with languages more than 04)
Screen Shot 2022-09-29 at 7 18 53 PM

Toolbar margins
Screen Shot 2022-09-29 at 7 21 39 PM

Resource components naming
Screen Shot 2022-09-29 at 7 17 21 PM
Screen Shot 2022-09-29 at 7 17 58 PM

### CODE REVIEW

File: src/components/interaction-designer/BuilderCanvas.vue [hunk 1]
```diff
@@ -5,9 +5,7 @@
     :style="{ minWidth: `${canvasWidth}px` , minHeight: `${canvasHeight}px` }">
     <block
       v-for="block in activeFlow.blocks"
-      :id="`block/${block.uuid}`"
```
Comment (safydy-r) [inline src/components/interaction-designer/BuilderCanvas.vue:8]: moved to Block.vue

File: src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue [hunk 2]
```diff
@@ -694,8 +704,8 @@ export default TreeBuilderToolbar
 .viamo-app-container .tree-builder-toolbar {
   position: sticky;
   left: 0;
-  padding-left: 100px;
-  width: calc(100vw - 30px);
+  padding-left: 94px;
+  width: calc(100vw - 5px);
```
Comment (safydy-r) [inline src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue:708]: these fix the toolbar margin pbm in voto5

File: src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue [hunk 3]
```diff
@@ -614,6 +614,16 @@ export class TreeBuilderToolbar extends mixins(Routes, Permissions, Lang) {
...
   }
 
+  handleFlowViewMenu() {
+    this.handlePersistFlow(this.treeUrl)
+  }
+
+  handleResourceViewMenu() {
+    this.handlePersistFlow(this.resourceUrl)
+    // scroll back to the top left in case the window has scroll bars from the builder-canvas
+    window.scrollTo(0, 0)
```
Comment (safydy-r) [inline src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue:624]: this fixes the scroll issue when we switch to resource viewer

File: src/views/InteractionDesigner.vue [hunk 4]
```diff
@@ -265,29 +265,30 @@ export class InteractionDesigner extends mixins(Lang, Routes) {
...
-
-    setTimeout(() => {
-      const {blockId, field} = this.$route.params
-      if (blockId) {
-        this.activateBlock({blockId})
-        scrollBlockIntoView(blockId)
-      }
-      if (field) {
-        scrollBehavior(this.$route)
+    if (this.isBuilderCanvasEnabled) {
```
Comment (safydy-r) [inline src/views/InteractionDesigner.vue:268]: Wrapped the old code inside  `if (this.isBuilderCanvasEnabled)`

File: src/store/validation/validationHelpers.ts [hunk 5]
```diff
@@ -99,8 +99,8 @@ function getErrorMessageLocalizationKeyForProperty(keyPrefix: string, ajvErrorOb
   }
 
   property = property.replaceAll('/', '-')
-    // Replacing digits to eliminate resource indexes
-    .replaceAll(/\d/g, 'x')
+    // Replacing digits to eliminate resource indexes, multiple digits number (Eg: 10, 11, etc) should be replaced with 01 'x'
+    .replace((property.match(/\d+/) ?? '') as string, 'x')
```
Comment (safydy-r) [inline src/store/validation/validationHelpers.ts:103]: this fixes the validation pbm which gave us `xx` for number `10` for eg. Now, with the fix, 10 will give one single `x`

File: src/store/flow/flow.ts [hunk 6]
```diff
@@ -351,7 +355,9 @@ export const actions: ActionTree<IFlowsState, IRootState> = {
 
   async flow_createBlankResourceForEnabledModesAndLangs({rootState, dispatch}): Promise<IResource> {
     // Let's create all languages, we might need them but if they are switched off they just don't get used
-    const values: IResourceValue = rootState.trees.ui.languages.reduce((memo: object[], language: { id: string, name: string }) => {
+    // It's important to use the same order to generate resource, that will be helpful for resource validation indexes
+    const values: IResourceValue[] = orderLanguages(rootState.trees.ui.languages as ILanguage[])
```
Comment (safydy-r) [inline src/store/flow/flow.ts:359]: this fixes the 2nd part of the validation pbm

**Review by object-required — APPROVED** [review]

File: src/store/validation/validationHelpers.ts (hunk 5 above)
Comment (object-required) [reply src/store/validation/validationHelpers.ts:103]: great observation 💯

File: src/store/flow/flow.ts [hunk 7]
```diff
@@ -23,6 +23,10 @@ import {ErrorObject} from 'ajv'
 import {IFlowsState} from '.'
 import {mergeFlowContainer} from './utils/importHelpers'
 
+export function orderLanguages(LanguagesList: ILanguage[]): ILanguage[] {
+  return sortBy(LanguagesList, ['label'])
```
Comment (bulatgab) [inline src/store/flow/flow.ts:27]: minor: use camelCase for the parameter

File: src/components/resource-editor/BatchMatchAudioFilesPrompt.vue [hunk 8]
```diff
@@ -1,209 +0,0 @@
...
-.batch-match-audio-files-prompt {
-  > p, > ul {
-    margin-bottom: 10px;
-  }
-
-  table {
-    background: white;
-  }
-}
-</style>
```
Comment (bulatgab) [inline src/components/resource-editor/BatchMatchAudioFilesPrompt.vue:209]: just confirming: this component wasn't used?

File: src/components/interaction-designer/resource-editors/PerLanguageResourceEditorRow.vue [hunk 9]
```diff
@@ -0,0 +1,95 @@
+<template>
+  <!--Resource editors grouped by language-->
+  <div v-if="resource"
+       class="language-resource-editor d-flex flex-column">
```
[the remark this answers is not in the document]
Comment (safydy-r) [reply src/components/interaction-designer/resource-editors/PerLanguageResourceEditorRow.vue:4]: good spot, I've fixed in the recent PR

**Pull request merged**